pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/DefaultRebalancePreChecker.java

```diff
@@ -92,17 +92,21 @@ public Map<String, RebalancePreCheckerResult> check(PreCheckContext preCheckCont
     // Check whether minimizeDataMovement is set in TableConfig
     preCheckResult.put(IS_MINIMIZE_DATA_MOVEMENT,
         checkIsMinimizeDataMovement(tableConfig, rebalanceConfig, tableRebalanceLogger));
+    // Determine the disk utilization threshold to use - either from rebalance config override or default
+    double diskUtilizationThreshold = rebalanceConfig.getDiskUtilizationThreshold() >= 0.0
+        ? rebalanceConfig.getDiskUtilizationThreshold() : _defaultDiskUtilizationThreshold;
+    // clip the disk utilization threshold to [0.0, 1.0]
+    diskUtilizationThreshold = Math.min(1.0, diskUtilizationThreshold);
```

Comment:
let's log a warning if we clip so users are aware in case they wonder why the value they provided changed